File: src/Engine/EmptyObject.js
import {Transform} from "./Transform.js";

export class EmptyObject {
    transform;

    constructor() {
        this.transform = new Transform();
    }

    get worldMatrix(){
        return this.transform.worldMatrix;
    }

    setPosition(v) {
        this.transform.position = v;
    }

    setRotation(v) {
        this.transform.rotation = v;
    }

    setScale(v) {
        this.transform.scale = v;
    }
}

File: CPP_engine/Engine/EmptyObject.cpp
#include "EmptyObject.h"

EmptyObject::EmptyObject(){
    this->transform = {};
}

Matrix4 EmptyObject::worldMatrix() {
    return this->transform.worldMatrix();
}

void EmptyObject::setPosition(Vector3 v) {
    this->transform.position = v;
}

void EmptyObject::setRotation(Vector3 v) {
    this->transform.rotation = v;
}

void EmptyObject::setScale(Vector3 v) {
    this->transform.scale = v;
}
